### PR TITLE
Refactor: introduce injectLocale

### PR DESCRIPTION
**Pull request by benjycui**
Close: #5103

### CODE REVIEW

**Comment by benjycui** [comment]
Modal and time-related components are a little difficult to refactor, I am going to leave them in this PR.

**Review by BANG88 — COMMENTED** [review]
These changes LGTM. I see the generated code also very clean.

File: components/auto-complete/index.tsx [hunk 1]
```diff
@@ -71,10 +71,6 @@ export default class AutoComplete extends React.Component<AutoCompleteProps, any
     showSearch: false,
   };
 
-  static contextTypes = {
```
Comment (benjycui) [inline components/auto-complete/index.tsx]: Useless code.

File: components/locale-provider/injectLocale.tsx [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+      static contextTypes = {
+        antLocale: PropTypes.object,
+      };
+
+      context: ComponentContext;
+
+      getLocale() {
+        const { antLocale } = this.context;
+        const localeFromContext = antLocale && antLocale[componentName];
+        const localeFromProps: any = this.props.locale || {};
```
Comment (benjycui) [inline components/locale-provider/injectLocale.tsx]: All the components that had been refactored with `injectLocale` will support `locale` property, but I am not going to document them, for the recommended way is `LocaleProvider`.

**Comment by benjycui** [comment]
@bang88 thx, and please check your Gmail.

File: components/pagination/Pagination.tsx [hunk 3]
```diff
@@ -1,87 +1,54 @@
...
+        selectComponentClass={isSmall ? MiniSelect : Select}
         locale={locale}
-        className={className}
       />
     );
   }
 }
+
+const injectPaginationLocale = injectLocale('Pagination', zhCN);
+export default injectPaginationLocale<PaginationProps>(Pagination as any);
```
Comment (yesmeck) [inline components/pagination/Pagination.tsx]: Use decorator?

Comment (benjycui) [inline components/pagination/Pagination.tsx]: I tried, but TypeScript stop me...

Comment (yesmeck) [inline components/pagination/Pagination.tsx]: 这有点蛋疼。。。

**Review by yesmeck — APPROVED** [review]

**Pull request merged**